wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1046,15 +1046,24 @@ public final class MLSService: MLSServiceInterface {
         context.performAndWait {
             let conversations = ZMConversation.fetchConversationsWithPendingProposals(in: context)
 
-            result = conversations.compactMap { conversation in
+            for conversation in conversations {
                 guard
                     let groupID = conversation.mlsGroupID,
                     let timestamp = conversation.commitPendingProposalDate
                 else {
-                    return nil
+                    continue
                 }
 
-                return (groupID, timestamp)
+                result.append((groupID, timestamp))
+
+                // The pending proposal might be for the subconversation,
+                // so include it just in case.
+                if let subgroupID = subconverationGroupIDRepository.fetchSubconversationGroupID(
```

Comment:
So we have no way of knowing if the proposal is for the subgroup or the parent group?

Reply:
We do, at the time we receive the message containing the pending proposal. If we really wanted to we could also persist this information such that at this later point in time we could know which of the two groups has proposals. But I couldn't see the benefit that would warrant the necessary overhead. Do you have something in mind?

Reply:
Thanks for the explanation, I wasn't quite sure what the whole flow looked like. I agree the benefit isn't worth the overhead